### PR TITLE
gpsbabel: fix darwin build

### PR DESCRIPTION
**Pull request by rycee**
###### Motivation for this change

To hopefully fix [build on Darwin](https://hydra.nixos.org/build/49778334):
```
clang  -g -O2 -Wall -c   -I. -F/nix/store/v8lv37s2yi2sz6h2w8nbcim9mdd2g0fn-qt-4.8.7/lib  -DHAVE_CONFIG_H -DNEW_STRINGS mac/libusb/darwin.c -o mac/libusb/darwin.o
mac/libusb/darwin.c:62:10: fatal error: 'IOKit/IOCFBundle.h' file not found
#include <IOKit/IOCFBundle.h>
```

Note, I have no means to build on OSX so cannot say whether this PR actually fixes the issue.

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by LnL7** [comment]
One of the tests failed

```
Running ./testo.d/tomtom_asc.test
--- ./reference/tomtom_poi.asc	2016-11-27 14:05:58.000000000 +0000
+++ /tmp/gpsbabel.29744/tomtom_poi.asc	2017-03-07 00:24:01.000000000 +0000
@@ -1,7 +1,7 @@
  TomTom Navigator Places of Interest
  GPSBabel- ASCII Export
   Points
- Created at: Thu Jan 01 00:00:00 1970
+ Created at: Thu M01 01 00:00:00 1970
 -71.119277,42.438878,"5066"
 -71.119689,42.439227,"5067"
 -71.116146,42.438917,"5096"
ERROR comparing ./reference/tomtom_poi.asc /tmp/gpsbabel.29744/tomtom_poi.asc
Running ./testo.d/tomtom_itn.test
```

**Comment by rycee** [comment]
That's a peculiar error. Looking at the gpsbabel source it seems to use
```C++
QDateTime dt = QDateTime::fromTime_t(time);
dt = dt.toTimeSpec(Qt::UTC);
QString dts = dt.toString("ddd MMM dd hh:mm:ss yyyy");
cout.replace("__DATE_AND_TIME__", dts);
```
and looking at the QT 4.8 documentation for [toString](https://doc.qt.io/qt-4.8/qdatetime.html#toString) it should interpret MMM as "the abbreviated localized month name (e.g. 'Jan' to 'Dec')" but it actually seems to translate only the last two Ms ("the month as number with a leading zero (01-12)").

Not sure what to make of this. Seems almost like a QT bug.

**Comment by matthewbauer** [comment]
Could you disable that one test? There's a precedent of just removing or commenting out that line in the test file with a patch. Alternatively, just set "doCheck" to ```!stdenv.isDarwin```.

**Comment by rycee** [comment]
Force pushed a commit which adds `if stdenv.isDarwin then "rm -v testo.d/tomtom_asc.test;". Hopefully it'll build OK with that.

File: pkgs/applications/misc/gpsbabel/default.nix [hunk 1]
```diff
@@ -43,6 +44,8 @@ stdenv.mkDerivation rec {
   '' + (
     # The raymarine and gtm tests fail on i686 despite -ffloat-store.
     if stdenv.isi686 then "rm -v testo.d/raymarine.test testo.d/gtm.test;"
+    # The tomtom asc test fails on darwin, see PR #23572.
+    else if stdenv.isDarwin then "rm -v testo.d/tomtom_asc.test;"
```
Comment (LnL7) [inline pkgs/applications/misc/gpsbabel/default.nix]: Using `lib.optionalString` here is a little bit nicer.

**Comment by LnL7** [comment]
The darwin stdenv now uses `llvm-4` which is a little bit more strict about pointer comparison.

```
clang++  -g -O2 -Wall -c   -I. -F/nix/store/szvcwp7q53aw0mzhvc60v4v6kcyzhpl7-qt-4.8.7/lib  -DHAVE_CONFIG_H -DNEW_STRINGS wintec_tes.cc -o wintec_tes.o
clang++  -g -O2 -Wall -c   -I. -F/nix/store/szvcwp7q53aw0mzhvc60v4v6kcyzhpl7-qt-4.8.7/lib  -DHAVE_CONFIG_H -DNEW_STRINGS subrip.cc -o subrip.o
bushnell.cc:138:36: error: ordered comparison between pointer and zero ('const char *' and 'int')
  for (t = bushnell_icons; t->icon > 0; t++) {
                           ~~~~~~~ ^ ~
bushnell.cc:150:36: error: ordered comparison between pointer and zero ('const char *' and 'int')
  for (t = bushnell_icons; t->icon > 0; t++) {
                           ~~~~~~~ ^ ~
2 errors generated.
make: *** [Makefile:128: bushnell.o] Error 1
make: *** Waiting for unfinished jobs....
builder for ‘/nix/store/kwh7xxq16h20f088pvdc9zsjkhqbsmsd-gpsbabel-1.5.3.drv’ failed with exit code 2
```

File: pkgs/applications/misc/gpsbabel/default.nix (hunk 1 above)
Comment (rycee) [inline pkgs/applications/misc/gpsbabel/default.nix]: Switched to `optionalString`.

**Comment by rycee** [comment]
Hmm, I won't attempt to fix the llvm-4 issue since it seems like it would need some more work and I'm unfortunately not having more time to spend.

**Comment by LnL7** [comment]
It should be pretty straight forward, I'll take a look.

**Comment by rycee** [comment]
Thanks! I think you should be able to push commits to my branch.

**Pull request merged**